node/src/main/scala/org/bitcoins/node/Node.scala

```diff
@@ -69,19 +69,16 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
     * the [[ChainApi chain api]] is updated inside of the p2p client
     */
   lazy val clientF: Future[P2PClient] = {
-    for {
-      chainApi <- chainApiFromDb()
-    } yield {
-      val peerMsgRecv: PeerMessageReceiver =
-        PeerMessageReceiver.newReceiver(chainApi = chainApi,
-                                        peer = peer,
-                                        callbacks = nodeCallbacks,
-                                        initialSyncDone = initialSyncDone)
-      val p2p = P2PClient(context = system,
-                          peer = peer,
-                          peerMessageReceiver = peerMsgRecv)
-      p2p
-    }
+    val chainApi = chainApiFromDb()
+    val peerMsgRecv: PeerMessageReceiver =
+      PeerMessageReceiver.newReceiver(chainApi = chainApi,
+                                      peer = peer,
+                                      callbacks = nodeCallbacks,
+                                      initialSyncDone = initialSyncDone)
+    val p2p = P2PClient(context = system,
+                        peer = peer,
+                        peerMessageReceiver = peerMsgRecv)
+    Future.successful(p2p)
   }
```

Comment:
Should remove the `Future` from this

Reply:
This will significantly add to the diff, i'm filing an issue for it

Reply:
#2262